ground/src/androidTest/java/com/google/android/ground/DataBindingIdlingResource.kt

```diff
@@ -0,0 +1,117 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground
+
+import android.R
+import android.view.View
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.fragment.app.FragmentActivity
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.IdlingResource
+import java.util.*
+
+/**
+ * Used to make Espresso work with DataBinding. Without it the tests will be flaky because
+ * DataBinding uses Choreographer class to synchronize its view updates hence using this to monitor
+ * a launched fragment in fragment scenario will make Espresso wait before doing additional checks
+ */
+class DataBindingIdlingResource : IdlingResource {
+  // Holds whether isIdle was called and the result was false. We track this to avoid calling
+  // onTransitionToIdle callbacks if Espresso never thought we were idle in the first place.
+  private var wasNotIdle = false
+  private var activity: FragmentActivity? = null
+  override fun getName(): String {
+    return String.format("DataBinding %s", ID)
+  }
+
+  override fun isIdleNow(): Boolean {
+    var idle = false
+    for (b in bindings) {
+      if (b == null) {
+        continue
+      }
+      if (!b.hasPendingBindings()) {
+        idle = true
+        break
+      }
+    }
+    if (idle) {
+      if (wasNotIdle) {
+        // Notify observers to avoid Espresso race detector.
+        for (cb in IDLING_CALLBACKS) {
+          cb.onTransitionToIdle()
+        }
+      }
+      wasNotIdle = false
+    } else {
+      wasNotIdle = true
+      // Check next frame.
+      if (activity != null) {
+        activity!!.findViewById<View>(R.id.content).postDelayed({ this.isIdleNow }, 16)
+      }
+    }
+    return idle
+  }
+
+  override fun registerIdleTransitionCallback(callback: IdlingResource.ResourceCallback) {
+    IDLING_CALLBACKS.add(callback)
+  }
+
+  /** Sets the activity from an [ActivityScenario] to be used from [DataBindingIdlingResource]. */
+  fun <T : FragmentActivity?> monitorActivity(activityScenario: ActivityScenario<T>) {
+    activityScenario.onActivity { activity: T -> this.activity = activity }
+  }
+
+  private fun getBinding(view: View?): ViewDataBinding? {
+    return DataBindingUtil.getBinding(view!!)
+  }
+
+  private val bindings: List<ViewDataBinding?>
+    get() {
+      val fragments =
+        if (activity == null) emptyList() else activity!!.supportFragmentManager.fragments
+      val bindings: MutableList<ViewDataBinding?> = ArrayList()
+      for (f in fragments) {
+        if (f.view == null) {
+          continue
+        }
```

Comment:
I think this might be easier to follow if we stored these views in local variables
```
val view = f.view ?: continue
```

Then the whole if statement can be collapsed and we can reference the new non null view variable below

Reply:
Done